spatialos-sdk/src/worker/connection.rs

```diff
@@ -328,30 +325,79 @@ impl Connection for WorkerConnection {
         }
     }
 
-    fn send_command_request(
+    fn send_command_request<C: Component>(
         &mut self,
-        _entity_id: EntityId,
-        _request: CommandRequest,
-        _timeout_millis: Option<u32>,
-        _command_parameters: CommandParameters,
+        entity_id: EntityId,
+        request: C::CommandRequest,
+        timeout_millis: Option<u32>,
+        allow_short_circuit: bool,
     ) -> RequestId<OutgoingCommandRequest> {
-        unimplemented!()
+        let timeout = match timeout_millis {
+            Some(c) => &c,
+            None => ptr::null(),
+        };
+
+        let command_request = Worker_CommandRequest {
+            reserved: ptr::null_mut(),
+            component_id: C::ID,
+            schema_type: ptr::null_mut(),
+            user_handle: component::handle_allocate(request),
+        };
+
+        let params = Worker_CommandParameters {
+            allow_short_circuit: allow_short_circuit as _,
+        };
+
+        let request_id = unsafe {
+            Worker_Connection_SendCommandRequest(
+                self.connection_ptr,
+                entity_id.id,
+                &command_request,
+                0,
```

Comment:
This shouldn't be 0. The command ID corresponds to the index of the command in the component (first command with ID 1). This would involve making a change to the generated code to add some kind of `get_command_id()` method onto the request and response type. I'll probably do this now quickly (as I have time).

Reply:
I pushed a commit which does this.